### PR TITLE
IG-2292 [Nuclio] Configuration: Add explanatory tooltips to panes

### PR DESCRIPTION
**Pull request by andrii-porokhnavets**

### CODE REVIEW

File: src/igz_controls/components/more-info/more-info.component.js [hunk 1]
```diff
@@ -6,8 +6,9 @@
             bindings: {
                 description: '@',
                 isDisabled: '<?',
+                trigger: '@?', // 'hover' or 'click'
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.component.js]: bindings description should be defined at the beginning of the file

File: src/igz_controls/components/more-info/more-info.component.js [hunk 2]
```diff
@@ -19,6 +20,8 @@
         var ctrl = this;
 
         ctrl.$onInit = onInit;
+        ctrl.onQuestionMarkClick = onQuestionMarkClick;
+        ctrl.isClickMode = isClickMode;
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.component.js:23]: sort functions

File: src/igz_controls/components/more-info/more-info.component.js [hunk 3]
```diff
@@ -30,11 +33,36 @@
         function onInit() {
             lodash.defaults(ctrl, {
                 isDisabled: false,
+                isHtmlEnabled: false,
                 isDefaultTooltipEnabled: false,
-                isDefaultTooltipHtml: false,
                 defaultTooltipPlacement: 'auto',
-                defaultTooltipPopupDelay: '0'
+                defaultTooltipPopupDelay: '0',
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.component.js]: remove trailing comma

File: src/igz_controls/components/more-info/more-info.component.js [hunk 4]
```diff
@@ -30,11 +33,36 @@
...
             });
+
+            // Defaults trigger method to 'mouseenter' (hover)
+            ctrl.trigger = ctrl.trigger === 'click' ? 'click' : 'mouseenter';
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.component.js]: add `ctrl.trigger` to `defaults` method above

File: src/igz_controls/components/more-info/more-info.tpl.html [hunk 5]
```diff
@@ -1,28 +1,39 @@
-<div class="more-info-wrapper">
-    <div data-ng-if="!$ctrl.isDefaultTooltipEnabled"
-         data-ng-class="{'disabled': $ctrl.isDisabled}"
-         class="question-mark igz-icon-help-round">
-    </div>
-    <div data-ng-if="!$ctrl.isDefaultTooltipEnabled"
-         class="row-description">
-        {{$ctrl.description}}
+<div class="more-info-wrapper" id="more-info-wrapper">
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.tpl.html]: remove id

File: src/igz_controls/components/more-info/more-info.tpl.html [hunk 6]
```diff
@@ -1,28 +1,39 @@
...
+    <div data-ng-if="!$ctrl.isDefaultTooltipEnabled">
+        <div data-ng-class="{'disabled': $ctrl.isDisabled, 'click-trigger': $ctrl.isClickMode()}"
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.tpl.html:3]: maybe you can use one-time function call

File: src/nuclio/projects/project/functions/version/version-configuration/tabs/version-configuration-build/version-configuration-build.tpl.html [hunk 7]
```diff
@@ -42,7 +48,7 @@
                 </igz-validating-input-field>
             </div>
             <div class="igz-col-50 build-field build-onbuild-image-field">
-                <div class="field-label">Onbuild image</div>
+                <div class="field-label label-with-tooltip align-items-center">Onbuild image</div>
```
Comment (mavdryk) [inline src/nuclio/projects/project/functions/version/version-configuration/tabs/version-configuration-build/version-configuration-build.tpl.html]: you added `label-with-tooltip` class, but where is tooltip?

File: src/nuclio/common/components/edit-item/edit-item.component.js [hunk 8]
```diff
@@ -918,6 +932,21 @@
             }
         }
 
+        /**
+         * Sets corresponding tooltip
+         */
+        function setTooltip() {
+            if (ctrl.selectedClass.id === 'secret') {
```
Comment (mavdryk) [inline src/nuclio/common/components/edit-item/edit-item.component.js]: add a map for tooltips

File: src/nuclio/common/components/edit-item/edit-item.component.js [hunk 9]
```diff
@@ -918,6 +932,21 @@
             }
         }
 
+        /**
+         * Sets corresponding tooltip
+         */
+        function setTooltip() {
```
Comment (mavdryk) [inline src/nuclio/common/components/edit-item/edit-item.component.js]: rename to `getTooltip` and delete from init and selectClass

**Review by mavdryk — CHANGES_REQUESTED** [review]

File: src/nuclio/projects/project/functions/version/version-configuration/tabs/version-configuration-build/version-configuration-build.tpl.html (hunk 7 above)
Comment (andrii-porokhnavets) [reply src/nuclio/projects/project/functions/version/version-configuration/tabs/version-configuration-build/version-configuration-build.tpl.html]: This is necessary for alignment

File: src/igz_controls/components/more-info/more-info.component.js [hunk 10]
```diff
@@ -36,18 +36,30 @@
...
-                defaultTooltipPopupDelay: '0',
+                defaultTooltipPopupDelay: '0'
             });
 
-            // Defaults trigger method to 'mouseenter' (hover)
-            ctrl.trigger = ctrl.trigger === 'click' ? 'click' : 'mouseenter';
-            ctrl.isDescriptionVisible = !isClickMode(); // need only for 'click' trigger. Init value - `false`
+            // Defaults trigger method to 'mouseenter'. Available 2 modes: `hover (mouseenter)` and `click`.
+            if (ctrl.trigger !== 'click') {
+                ctrl.trigger = 'mouseenter';
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.component.js:44]: move this to the `defaults` method above

File: src/igz_controls/components/more-info/more-info.component.js [hunk 11]
```diff
@@ -36,18 +36,30 @@
...
+            }
+
+            // If `click` mode - init value is `false` and will trigger on click. If `hover` mode - always `true`.
```
Comment (mavdryk) [inline src/igz_controls/components/more-info/more-info.component.js]: write a more descriptive comment, what this variable does in each mode

File: src/nuclio/common/components/edit-item/edit-item.component.js [hunk 12]
```diff
@@ -445,6 +434,30 @@
             return ctrl.type === 'volume' ? ctrl.item.volume.name : ctrl.item.name;
         }
 
+        /**
+         * Gets corresponding tooltip description
+         * @returns {string}
+         */
+        function getTooltip() {
+            var tooltips = {
```
Comment (mavdryk) [inline src/nuclio/common/components/edit-item/edit-item.component.js]: add this variable to `ctrl.tooltips`. You don't need to define it on each function call

File: src/nuclio/projects/project/functions/version/version-configuration/tabs/version-configuration-build/version-configuration-build.tpl.html (hunk 7 above)
Comment (mavdryk) [reply src/nuclio/projects/project/functions/version/version-configuration/tabs/version-configuration-build/version-configuration-build.tpl.html]: But there isn't any tooltip, maybe `align-items-center` will be enough for alignment?

File: src/nuclio/common/components/edit-item/edit-item.component.js [hunk 13]
```diff
@@ -573,6 +590,14 @@
             return ctrl.selectedClass.id === 'cron';
         }
 
+        /**
+         * Checks if tooltip is visible.
+         * @returns {boolean} `true` only for `secret` and `configMap` classes of volume type, `false` otherwise.
+         */
+        function isTooltipVisible() {
+            return lodash.includes(['secret', 'configMap'], ctrl.selectedClass.id);
```
Comment (mavdryk) [inline src/nuclio/common/components/edit-item/edit-item.component.js]: it's better to use classes from the map that you created above. Therefore you will not have duplication of classes

File: src/nuclio/projects/project/functions/functions.service.js [hunk 14]
```diff
@@ -450,11 +450,13 @@
                     },
                     {
                         id: 'secret',
-                        name: 'Secret'
+                        name: 'Secret',
+                        tooltip: 'Managing sensitive objects'
```
Comment (mavdryk) [inline src/nuclio/projects/project/functions/functions.service.js:454]: you don't use this property `tooltip`, why do you need it?

File: src/nuclio/projects/project/functions/functions.service.js [hunk 15]
```diff
@@ -450,11 +450,13 @@
...
                     },
                     {
                         id: 'configMap',
-                        name: 'ConfigMap'
+                        name: 'ConfigMap',
+                        tooltip: 'Storing configuration'
```
Comment (mavdryk) [inline src/nuclio/projects/project/functions/functions.service.js:459]: you don't use this property `tooltip`, why do you need it?

**Review by mavdryk — CHANGES_REQUESTED** [review]

File: src/nuclio/projects/project/functions/functions.service.js (hunk 15 above)
Comment (andrii-porokhnavets) [reply src/nuclio/projects/project/functions/functions.service.js:459]: This property is used for tooltip in dropdown

File: src/nuclio/projects/project/functions/functions.service.js (hunk 14 above)
Comment (andrii-porokhnavets) [reply src/nuclio/projects/project/functions/functions.service.js:454]: This property is used for tooltip in dropdown

File: src/igz_controls/components/more-info/more-info.component.js (hunk 10 above)
Comment (andrii-porokhnavets) [reply src/igz_controls/components/more-info/more-info.component.js:44]: for what?

**Pull request closed without merging**